service/src/main/kotlin/io/provenance/explorer/web/pulse/PulseAssetController.kt

```diff
@@ -0,0 +1,94 @@
+import io.swagger.v3.oas.annotations.tags.Tag
+import org.jetbrains.exposed.sql.SortOrder
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.math.BigDecimal
+
+@RestController
+@RequestMapping(
+    path = ["/api/pulse/asset"],
+    produces = [MediaType.APPLICATION_JSON_VALUE]
+)
+@Tag(
+    name = "Pulse Assets",
+    description = "Pulse asset endpoint"
+)
+class PulseAssetController(private val pulseMetricService: PulseMetricService) {
+
+    @Operation(summary = "Exchange-traded Asset Summaries")
+    @GetMapping("/summary/list")
+    fun getPulseAssetSummaries(
+        @RequestParam(required = false) search: String?,
+                               @RequestParam(required = false) sortOrder: List<SortOrder>?,
+                               @RequestParam(required = false) sortColumn: List<String>?
+    ): List<PulseAssetSummary> =
+        pulseMetricService.pulseAssetSummaries().filter {
+            // this is a small list so we can get away with this
+            search.isNullOrBlank() ||
+                    it.name.contains(search, ignoreCase = true) ||
+                    it.symbol.contains(search, ignoreCase = true) ||
+                    it.display.contains(search, ignoreCase = true) ||
+                    it.base.contains(search, ignoreCase = true) ||
+                    it.description.contains(search, ignoreCase = true)
+        }.sortedWith(
+            Comparator { a, b ->
+            if (sortColumn == null || sortOrder == null) return@Comparator 0
+
+            for (i in sortColumn.indices) {
+                val column = sortColumn[i]
+                val order = sortOrder[i]
+
+                val comparisonResult = when (column) {
+                    "name" -> compareValues(
+                        a.name.ifEmpty { a.base },
+                        b.name.ifEmpty { b.base }
+                    )
+                    "price" -> compareValues(a.priceTrend?.currentQuantity ?: BigDecimal.ZERO, b.priceTrend?.currentQuantity ?: BigDecimal.ZERO)
+                    "marketCap" -> compareValues(a.marketCap, b.marketCap)
+                    "volume" -> compareValues(a.volumeTrend?.currentQuantity ?: BigDecimal.ZERO, b.volumeTrend?.currentQuantity ?: BigDecimal.ZERO)
+                    else -> throw IllegalArgumentException("Invalid sort column: $column")
+                }
+
+                if (comparisonResult != 0) {
+                    return@Comparator if (order == SortOrder.ASC) comparisonResult else -comparisonResult
+                }
+            }
+            0
+        }
+        )
+
+    /**
+     * Note that `denom` is a required request param instead of a path variable
+     * because denoms like `nft/blahblahblah` exist and are not valid path variables.
```

Comment:
see also: all the ibc denoms... these caused the REST apis to break a long time ago...